src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -310,6 +323,40 @@
     <MakeDir Directories="$(ProjectSpecificConfDir)" />
   </Target>
 
+  <!-- **************************************************************************** -->
+  <!-- In case of Razor compilation the protobuf files generated during the core compilation step are already moved into a temporary folder.
+       This target moves the protobuf files generated during the razor core compile step into a .razor directory, and alsomoves the protobuf files
+       from the temporary directory back to their original place. This target also takes care about generating the ProjectInfo.xml file for the razor part of the compilation.
+  -->
+  <!-- **************************************************************************** -->
```

Comment:
Id sort of has... The TemporaryProjectSpecificOutDir property will be set only in case of a razor build.

Reply:
That's kinda hidden dependency, because `TemporaryProjectSpecificOutDir` doesn't tell anything about the Razor stuff. In theory, we could add Razor, Blazor, Crazor, Weirdzor in the future that will need different directory names :) 

Let's add `$(RazorCompileOnBuild) == 'true' AND ...`